plugin/helpers_plugin_test.go

```diff
@@ -103,3 +104,30 @@ func TestInstallPluginFromURL(t *testing.T) {
 		assert.Equal(t, "received 404 status code while downloading plugin from server", err.Error())
 	})
 }
+
+func TestGetPluginAssetURL(t *testing.T) {
+	siteURL := "https://mattermost.example.com"
+	api := &plugintest.API{}
+	api.On("GetConfig").Return(&model.Config{ServiceSettings: model.ServiceSettings{SiteURL: &siteURL}})
+
+	p := &plugin.HelpersImpl{API: api}
+
+	t.Run("Valid directory and pluginID provided", func(t *testing.T) {
+		pluginID := "mattermost-1234"
+		dir := "assets"
+		wantedURL := &url.URL{Scheme: "https", Host: "mattermost.example.com", Path: "/mattermost-1234/assets"}
+		gotURL, err := p.GetPluginAssetURL(pluginID, dir)
+
+		assert.Equalf(t, wantedURL, gotURL, "GetPluginAssetURL(%s, %s) got=%s; want=%v", pluginID, dir, gotURL, wantedURL)
+		assert.NoError(t, err)
+	})
+
+	t.Run("Invalid directory and pluginID provided", func(t *testing.T) {
```

Comment:
Should we add separate tests cases for invalid directory and invalid pluginID? (And then, based on my comment above, for when SiteURL is empty?)

Reply:
@lieut-data What criteria make a directory and pluginID invalid? can you please talk a bit on this?

Reply:
I was thinking specifically the case of `len(pluginID) == 0` and `len(asset) == 0`, vs. testing them both at the same time, when we're really only hitting one branch of the if condition.

Reply:
I have implemented all the feedback.